src/H5VLcallback.c

```diff
@@ -5875,6 +7158,50 @@ H5VLobject_optional(void *obj, hid_t connector_id, H5VL_object_optional_t opt_ty
     FUNC_LEAVE_API_NOINIT(ret_value)
 } /* end H5VLobject_optional() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5VLobject_optional_vararg
+ *
+ * Purpose:     Performs an optional connector-specific operation on an object
+ *
+ * Note:        Same as H5VLobject_optional, but uses varargs instead of a va_list
+ *
+ * Return:      Success:    Non-negative
+ *              Failure:    Negative
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5VLobject_optional_vararg(void *obj, hid_t connector_id, H5VL_object_optional_t opt_type, hid_t dxpl_id,
+                           void **req /*out*/, ...)
+{
+    H5VL_class_t *cls;                   /* VOL connector's class struct */
+    va_list       arguments;             /* Argument list passed from the API call */
+    hbool_t       arg_started = FALSE;   /* Whether the va_list has been started */
+    herr_t        ret_value   = SUCCEED; /* Return value */
+
+    FUNC_ENTER_API_NOINIT
+    H5TRACE5("e", "*xiVyix", obj, connector_id, opt_type, dxpl_id, req);
+
+    /* Check args and get class pointer */
+    if (NULL == obj)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid object")
+    if (NULL == (cls = (H5VL_class_t *)H5I_object_verify(connector_id, H5I_VOL)))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADTYPE, FAIL, "not a VOL connector ID")
+
+    /* Call the corresponding internal VOL routine */
+    HDva_start(arguments, req);
+    arg_started = TRUE;
+    if (H5VL__object_optional(obj, cls, opt_type, dxpl_id, req, arguments) < 0)
+        HGOTO_ERROR(H5E_VOL, H5E_CANTOPERATE, FAIL, "unable to execute object optional callback")
+
+done:
+    /* End access to the va_list, if we started it */
+    if (arg_started)
+        HDva_end(arguments);
+
+    FUNC_LEAVE_API_NOINIT(ret_value)
+} /* end H5VLobject_optional_vararg() */
+
```

Comment:
No H5VLobject_optional_op()?

Reply:
The dynamic operations are registered on a subclass type, so it didn't seem like a good fit to add the generic form (H5VLobject_optional_op).